client/post-editor/controller.js

```diff
@@ -107,11 +108,45 @@ function getPressThisContent( query ) {
 	return pieces.join( '\n\n' );
 }
 
+// TODO: REDUX - remove flux actions when whole post-editor is reduxified
+// Until the full migration, the Copy Post functionality needs to dispatch both Flux and Redux actions:
+// - (Flux) startEditingNew: to set the editor content;
+// - (Redux) editPost: to set every other attribute (in particular, to update the Category Selector, terms can only be set via Redux);
+// - (Flux) edit: to reliably show the updated post attributes before (auto)saving.
+function startEditingPostCopy( siteId, postToCopyId, context ) {
+	wpcom.site( siteId ).post( postToCopyId ).get().then( postToCopy => {
+		const postAttributes = pick(
+			postToCopy,
+			'canonical_image',
+			'content',
+			'excerpt',
+			'featured_image',
+			'format',
+			'metadata',
+			'post_thumbnail',
+			'terms',
+			'title',
+			'type'
+		);
+		postAttributes.tags = map( postToCopy.tags, 'name' );
+
+		actions.startEditingNew( siteId, {
+			type: 'post',
+			content: postToCopy.content,
+		} );
+		context.store.dispatch( editPost( siteId, null, postAttributes ) );
+		actions.edit( postAttributes );
+	} ).catch( () => {
+		actions.startEditingNew( siteId, { type: 'post' } );
+	} );
+}
+
 module.exports = {
 
 	post: function( context ) {
 		const postType = determinePostType( context );
 		const postID = getPostID( context );
+		const postToCopyId = context.query.copy || false;
```

Comment:
>We don't, but I thought it would make more obvious that it's a completely optional parameter.

If anything, it'd probably lead me to believe there is special consideration to be had in treating it as a boolean when the query value doesn't exist. In almost all cases, we're happy to leverage [JavaScript falsiness](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) in its simplest forms vs. explicitly handling specific values.